packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -303,6 +303,97 @@ export interface UserInvitationConfig {
   readonly smsMessage?: string;
 }
 
+/**
+ * The different ways in which a user pool's MFA enforcement can be configured.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-mfa.html
+ */
+export enum MfaEnforcement {
+  /** Users are not required to use MFA for sign in, and cannot configure one. */
+  OFF = 'OFF',
+  /** Users are not required to use MFA for sign in, but can configure one if they so choose to. */
+  OPTIONAL = 'OPTIONAL',
+  /** Users are required to configure an MFA, and have to use it to sign in. */
+  REQUIRED = 'ON',
+}
+
+/**
+ * The different ways in which a user pool can obtain their MFA token for sign in.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-mfa.html
+ */
+export interface MfaTypes {
+  /**
+   * The MFA token is sent to the user via SMS to their verified phone numbers
+   * @default true
+   */
+  readonly sms: boolean;
+
+  /**
+   * The MFA token is a time-based one time password that is generated by a hardware or software token
+   * @default false
+   */
+  readonly oneTimePassword: boolean;
```

Comment:
If `sms` is an initialism, should this be `otp`? (I think it's a widely-enough used term to be OK)

Reply:
I think the term they use in the cognito docs is TOTP...

Reply:
I started with TOTP but I thought it was a bit too cryptic. I can switch over to `otp` - it's better recognized than `totp`.